pkg/source/client.go

```diff
@@ -65,8 +66,22 @@ func NewConfig(ctx context.Context) ([]string, *sarama.Config, error) {
 
 	if env.Net.SASL.Enable {
 		cfg.Net.SASL.Enable = true
+		cfg.Net.SASL.Handshake = true
 		cfg.Net.SASL.User = env.Net.SASL.User
 		cfg.Net.SASL.Password = env.Net.SASL.Password
+
+		// We default to plain sasl type
+		cfg.Net.SASL.Mechanism = sarama.SASLTypePlaintext
+
+		if env.Net.SASL.Type == sarama.SASLTypeSCRAMSHA256 {
+			cfg.Net.SASL.SCRAMClientGeneratorFunc = func() sarama.SCRAMClient { return &XDGSCRAMClient{HashGeneratorFcn: SHA256} }
+			cfg.Net.SASL.Mechanism = sarama.SASLTypeSCRAMSHA256
+		}
+
+		if env.Net.SASL.Type == sarama.SASLTypeSCRAMSHA512 {
+			cfg.Net.SASL.SCRAMClientGeneratorFunc = func() sarama.SCRAMClient { return &XDGSCRAMClient{HashGeneratorFcn: SHA512} }
+			cfg.Net.SASL.Mechanism = sarama.SASLTypeSCRAMSHA512
+		}
```

Comment:
Seeing this made me realize we need validation for the type.

If the user makes a typo in the YAML, the code will start using `sarama.SASLTypePlaintext` but user won't notice that.

What do you think? Just validate 3 possible options in the webhook and reject the CR if the value is anything else?
I wouldn't perhaps create the webhook for this, but the webhook is already there.

In the webhook,

Reply:
oh, yes - good catch !

Reply:
Perhaps I'm misreading the code, but, do we have a validating webhook for the source bits?
https://github.com/knative-sandbox/eventing-kafka/blob/master/cmd/webhook/main.go#L49

lets do this in a later PR please

Reply:
@lberk feel free to "unhold" when ready

Reply:
Nah, it is here: https://github.com/knative-sandbox/eventing-kafka/blob/82c096577d36be37a12624367b0ebc96b9d9506e/cmd/source/controller/main.go#L194

The source has the controller and the webhook in the same pod.

But, +1 on merging w/o it and doing as follow up though.